### PR TITLE
PR for initial operator review

### PR DESCRIPTION
**Pull request by jeremyary**
@tchughesiv @jianrongzhang89 @redhatHameed Please use this PR as a chance to capture thoughts about the new operator introduced via this repo. Since this was our first push, it went to main, so I figure this is a good way to allow for feedback. The 'files changed' on here isn't going to reflect all the operator files, there were a lot of commits, so please explore the repo main branch and share feedback/thoughts here. Thanks!

### CODE REVIEW

**Comment by jeremyary** [comment]
cc @xieshenzh

File: README.md [hunk 1]
```diff
@@ -1 +1 @@
-# rds-dbaas-operator
```
[the remark this answers is not in the document]
Comment (redhatHameed) [reply README.md:1]: something like :   ```replicas: 1
  updatedReplicas: 1
  unavailableReplicas: 1```

Comment (redhatHameed) [inline README.md:1]: the inventory status showing if you added wrong keys: Is that possible change message ?
```

status:
  conditions:
    - lastTransitionTime: '2022-06-13T20:50:14Z'
      message: Installing RDS controller
      reason: Updating
      status: 'False'
      type: SpecSynced
    - lastTransitionTime: '2022-06-13T20:50:15Z'
      message: DBaaS Provider Custom Resource reconciliation in progress
      reason: ProviderReconcileInprogress
      status: 'False'
      type: InventoryReady
```

**Comment by jianrongzhang89** [comment]
@jeremyary @xieshenzh I could not run the unit test. I got errors like:
Failure [0.156 seconds]
[BeforeSuite] BeforeSuite 
/Users/zhangj/devt/src/rds-dbaas-operator/controllers/suite_test.go:83

  Unexpected error:
      <*fmt.wrapError | 0xc000698320>: {
          msg: "unable to start control plane itself: failed to start the controlplane. retried 5 times: exec: \"etcd\": executable file not found in $PATH",
          err: <*fmt.wrapError | 0xc000698300>{
              msg: "failed to start the controlplane. retried 5 times: exec: \"etcd\": executable file not found in $PATH",
              err: <*exec.Error | 0xc0006982c0>{
                  Name: "etcd",
                  Err: <*errors.errorString | 0xc000088390>{
                      s: "executable file not found in $PATH",
                  },
              },
          },
      }
      unable to start control plane itself: failed to start the controlplane. retried 5 times: exec: "etcd": executable file not found in $PATH
  occurred

Is there any special setting needed?

**Comment by jianrongzhang89** [comment]
While bundle-push is included in the release-build target?

**Comment by xieshenzh** [comment]
> While bundle-push is included in the release-build target?

Because `catalog-build` pulls the bundle image.

**Comment by xieshenzh** [comment]
https://github.com/RHEcosystemAppEng/rds-dbaas-operator/pull/1#discussion_r896128826

I don't think it is possible. The error details are only available in the logs of the ack pod. The ack Deployment only shows the minimum ready replicas requirement is not met, and the Deployment is updating.

**Pull request closed without merging**